sdks/java/core/src/test/java/org/apache/beam/sdk/runners/dataflow/TestCountingSource.java

```diff
@@ -164,21 +164,20 @@ public CountingSourceReader(int startingPoint) {
 
     @Override
     public boolean start() {
-      return true;
+      return advance();
     }
 
     @Override
     public boolean advance() {
-      if (current < numMessagesPerShard - 1) {
+      if (current >= numMessagesPerShard) {
+        return false;
+      }
         // If testing dedup, occasionally insert a duplicate value;
         if (dedup && ThreadLocalRandom.current().nextInt(5) == 0) {
           return true;
         }
         current++;
-        return true;
-      } else {
-        return false;
-      }
+      return current < numMessagesPerShard;
```

Comment:
the indentation looks off here, but maybe it's a tabs-vs-spaces thing? Except we shouldn't have tabs.. hmm.. can you please investigate?